### PR TITLE
compaction pre-and-post active row test

### PR DESCRIPTION
**Pull request by chessai**

### CODE REVIEW

File: test/Chainweb/Test/Pact/PactSingleChainTest.hs [hunk 1]
```diff
@@ -12,6 +12,8 @@
 {-# LANGUAGE TypeFamilies #-}
 {-# LANGUAGE ViewPatterns #-}
 
+{-# options_ghc -fno-warn-unused-imports #-}
```
Comment (chessai) [inline test/Chainweb/Test/Pact/PactSingleChainTest.hs:15]: remove this

File: src/Chainweb/Pact/Backend/Compaction.hs [hunk 2]
```diff
@@ -70,7 +73,7 @@ import Data.LogMessage
 import Pact.Types.SQLite (SType(..), RType(..))
 import Pact.Types.SQLite qualified as Pact
 
-newtype TxId = TxId Int64
+newtype ITxId = ITxId Int64
```
Comment (edmundnoble) [inline src/Chainweb/Pact/Backend/Compaction.hs:76]: Whence?

File: src/Chainweb/Pact/Backend/Compaction.hs [hunk 3]
```diff
@@ -626,3 +635,56 @@ compactMain = do
...
+  let excludeThese = checkpointerTables ++ compactionTables
+  let fmtTable x = "\"" <> x <> "\""
+
+  tables <- fmap sortedTableNames $ do
+    let qry =
+          "SELECT name FROM sqlite_schema \
+          \WHERE \
+          \  type = 'table' \
+          \AND \
+          \  name NOT LIKE 'sqlite_%'"
```
Comment (edmundnoble) [inline src/Chainweb/Pact/Backend/Compaction.hs:652]: I like that we have a blacklist for non-compacted tables rather than the other way around.

**Review by edmundnoble — APPROVED** [review]

File: src/Chainweb/Pact/Backend/Compaction.hs (hunk 2 above)
Comment (chessai) [reply src/Chainweb/Pact/Backend/Compaction.hs:76]: there is another type in scope that represents a TxId, and it's incompatible with this one. The 'I' means 'Int'. The reason this type exists is to provide some type-safety for TxIds used with Pact's SQLite API.

File: src/Chainweb/Pact/Backend/Compaction.hs (hunk 3 above)
Comment (chessai) [reply src/Chainweb/Pact/Backend/Compaction.hs:652]: So, you are saying, you prefer the code as written? Or are you suggesting a change?

**Pull request merged**